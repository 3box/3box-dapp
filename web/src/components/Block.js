import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

export function Block(props) {
  const {
    intl,
    web3 = null,
  } = props;
  const web3Status = web3 ? 'Connected' : 'NotConnected';
  return (
    <header className="App-header">
      <h1 className="App-title">
        {intl.formatMessage({ id: web3Status })}
      </h1>
    </header>
  );
}

Block.propTypes = {
  web3: PropTypes.object,
  intl: PropTypes.object.isRequired,
};

Block.defaultProps = {
  web3: null,
};

export default injectIntl(Block);
