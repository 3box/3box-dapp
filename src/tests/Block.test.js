import React from 'react';
import ReactDOM from 'react-dom';
import Block from '../components/Block';
import {
  renderWithIntl,
  rendererCreateWithIntl,
} from './helper';

const mockPropsNoConnection = {
  web3: null,
};

const mockPropsConnected = {
  web3: {},
};

test('renders without crashing', () => {
  const div = document.createElement('div');
  renderWithIntl(<Block {...mockPropsNoConnection} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('render correctly when no web3 connection', () => {
  const component = rendererCreateWithIntl(<Block {...mockPropsNoConnection} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with web3 connection', () => {
  const component = rendererCreateWithIntl(<Block {...mockPropsConnected} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
