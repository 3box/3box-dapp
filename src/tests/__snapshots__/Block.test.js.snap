// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly when no web3 connection 1`] = `
<header
  className="App-header"
>
  <h1
    className="App-title"
  >
    Not connect to the blockchain
  </h1>
</header>
`;

exports[`render correctly with web3 connection 1`] = `
<header
  className="App-header"
>
  <h1
    className="App-title"
  >
    Connected to the blockchain
  </h1>
</header>
`;
